src/compiler/reverse.jl

```diff
@@ -26,13 +26,28 @@ end
 unwrapquote(x) = x
 unwrapquote(x::QuoteNode) = x.value
 
-is_literal_getproperty(ex) =
-  (iscall(ex, Base, :getproperty) || iscall(ex, Core, :getfield) || iscall(ex, Base, :getfield)) &&
-  ex.args[3] isa Union{QuoteNode,Integer}
+is_getproperty(ex) = iscall(ex, Base, :getproperty)
 
 function instrument_getproperty!(ir, v, ex)
-  is_literal_getproperty(ex) ?
-    (ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], Val(unwrapquote(ex.args[3])))) :
+  if is_getproperty(ex)
+    if ex.args[3] isa Union{QuoteNode,Integer}
+      ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], Val(unwrapquote(ex.args[3])))
+    else
+      f = insert!(ir, v, :(Val($(ex.args[3]))))
+      ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], f)
+    end
+  else
+    ex
+  end
+end
+
+is_literal_getfield(ex) =
```

Comment:
to match `get_property` this should just be called `is_getfield`

Reply:
They are checking slightly different things though. `is_literal_getfield` also checks if the second argument is either a symbol or int literal.

Reply:
Do we not want to do this for if it is `getfield` via a variable or an expression also though?

Reply:
We could, but not sure we want to. Let's discuss that in the other thread: https://github.com/FluxML/Zygote.jl/pull/848#discussion_r542334717